### DISCUSSION TITLE
GitHub page blank after deploy: Failed to load resource: the server responded with a status of 404

### DISCUSSION THREAD

**Discussion by jiayibao** [community/community#23058] [Other Feature Feedback, Questions, & Ideas]
<p>Hi All, </p>
<p>I tried to build a personal website following a template and the website worked perfectly locally. However, after I deployed it to GitHub Pages, I can only see a blank page. I checked the Javascript console and here is the error message: </p>
<p><a href="https://jiayibao.github.io/https://jiayibao.github.io/personal-site//main-1d41357ad301d064dcf9.min.js" rel="nofollow noopener">https://jiayibao.github.io/https://jiayibao.github.io/personal-site//main-1d41357ad301d064dcf9.min.js </a></p>
<p>Failed to load resource: the server responded with a status of 404 ()</p>
<p>Does anyone have any clue why “<a href="https://jiayibao.github.io" rel="nofollow noopener">https://jiayibao.github.io</a>” is repeated twice in loading the resource? Here is my github repository: <a href="https://github.com/jiayibao/personal-site/" rel="nofollow noopener">https://github.com/jiayibao/personal-site/</a></p>
<p>And here is the personal webpage that is currently showing blank: <a href="https://jiayibao.github.io/personal-site/" rel="nofollow noopener">https://jiayibao.github.io/personal-site/</a></p>
<p>My suspect that the error has something to do with path, but I’m not really sure about how to fix it. Any help would be greatly appreciated! Thank you!</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/jiayibao">@jiayibao</a>,</p>
<p>Thanks for being here! To have your repository show up at <a href="https://username.github.io/your-repository" rel="nofollow noopener">https://username.github.io/your-repository</a>, you need to setup GitHub Pages for that repository as well. Ie: your repo should be <em>USERNAME</em>.github.io and its URL  ( <a href="https://github.com/USERNAME/USERNAME.github.io" rel="nofollow noopener">https://github.com/USERNAME/USERNAME.github.io</a> ).</p>
<p>I hope this helps!</p>

**Comment by jiayibao** [comment, marked as answer]
<p>Thanks! In fact, this problem has been solved by the person who provided me the inital website template. Thanks for the input! </p>

**Reply by ashishpandey17360** [reply]
Hey can you help me i don't understand?

**Reply by ID24-CMR** [reply]
??????????????????????
how does it solved our problem ?????

**Reply by DevHanza** [reply]
> ?????????????????????? how does it solved our problem ?????

Check here: https://github.com/orgs/community/discussions/23058#discussioncomment-3368631

**Comment by iamtanaybanerjee** [comment]
<p>hey, please help me as I am facing the same problem as yours.</p>

**Reply by Chauhdri** [reply]
I have changed private to false in Package.json. This change loads my website instead of blank page.

**Reply by CASTVELL** [reply]
Aleluya, thank u so much

**Reply by vovka8101** [reply]
two hours of suffering and I finally fixed that, thanks

**Reply by Ratnesh125** [reply]
thank you so much

**Reply by duynguyen224** [reply]
many thanks, it save my life

**Reply by PriDebnath** [reply]
> I have changed private to false in Package.json. This change loads my website instead of blank page.

Thank you @Chauhdri

**Comment by tehceen** [comment]
<p>thanks alot you saved my life.</p>

**Comment by rajatrai101** [comment]
<p>Can you please share the solution with us? Even though my name is not getting repeated in the HTTP request link but I feel I am facing a similar issue as you did. :slight_smile:</p>

**Comment by meghanar121** [comment]
<p>I am also facing same problem, this is my website in react <a href="https://meghanar121.github.io/covid-19/" rel="nofollow noopener">https://meghanar121.github.io/covid-19/</a></p>

**Comment by sjPrajwal** [comment]
<p><a class="mention" href="/meghanar121">@meghanar121</a> can you please suggest solution for this…? even i am facing same problem…</p>

**Comment by meghanar121** [comment]
<p><a class="mention" href="/sjprajwal">@sjPrajwal</a>  this was react app i made mistake in package.json file, homepage should have proper address<br>
you can follow steps for deploying react app here:<a href="https://github.com/gitname/react-gh-pages/tree/master" rel="nofollow noopener">https://github.com/gitname/react-gh-pages/tree/master</a></p>

**Reply by lwardlegit** [reply]
yea I had the same issue had to follow this tutorial thanks :)

**Comment by Shivaratnakumar** [comment]
<p><a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a> GitHub Pages is already configured. But it is showing loading error.</p>
<p>PF the below attachments.<br>
hosted site<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/aadee4114a3cb98e249f996603091f5d1ba8452b.png" data-download-href="/uploads/short-url/onAI7192NaxgQqJH22RngXrZkNl.png?dl=1" title="1" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/22994982/181074618-53152f27-f157-486f-827f-34eed743ee9b.png" alt="1" data-base62-sha1="onAI7192NaxgQqJH22RngXrZkNl" width="690" height="289" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/aadee4114a3cb98e249f996603091f5d1ba8452b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">1</span><span class="informations">1366×574 51.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><p></p>

**Comment by Shivaratnakumar** [comment]
<p><a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a>  github page configuration.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/ada4a7034f84adb7ac0371173df4cfdb3a98422c.png" data-download-href="/uploads/short-url/oM7mQzyh5zOjfNQrXcObOnMbg0A.png?dl=1" title="2" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/22994982/181074622-5d8bf2a8-c39a-4828-bde9-922f4dc0b0a3.png" alt="2" data-base62-sha1="oM7mQzyh5zOjfNQrXcObOnMbg0A" width="690" height="386" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/ada4a7034f84adb7ac0371173df4cfdb3a98422c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">2</span><span class="informations">915×513 18.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><p></p>

**Comment by oscarsinuco** [comment]
<p>I resolved that problem, try it: go to index.html in your compiled code, for example in Angular is “dist/index.html” and paste that:  if you already have it, then delete and replace that line.</p>

**Comment by lzmunch** [comment]
<p>Not sure why, but this worked for me: setting the GitHub Pages source to a different branch and then setting it back to the gh-pages branch (or whatever branch you deployed to).</p>

**Comment by GajananVS** [comment]
<p>Hi<br>
I am getting the same error in my angular app, can you please provide the step to resolve it ?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b295aca6e9217238bad9d3baa76b78242a9ae4ec.png" data-download-href="/uploads/short-url/ptPFgZhSye96DTHnGfSIOMNpzDC.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/44489961/181074630-2bd5d46a-e7d1-4c04-ac2f-cd2439d3a563.png" alt="image" data-base62-sha1="ptPFgZhSye96DTHnGfSIOMNpzDC" width="690" height="199" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b295aca6e9217238bad9d3baa76b78242a9ae4ec_2_690x199.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/b295aca6e9217238bad9d3baa76b78242a9ae4ec_2_1035x298.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/b295aca6e9217238bad9d3baa76b78242a9ae4ec.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b295aca6e9217238bad9d3baa76b78242a9ae4ec_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1363×394 24.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>application link<br>
<a href="https://gajananvs.github.io/Demo/" rel="nofollow noopener">https://gajananvs.github.io/Demo/</a></p>
<p>Thanks<br>
Gajanan</p>

**Comment by otter-computer** [comment]
<p>This looks like it might be a case sensitivity issue. Your Pages site is hosted from a repository called <code>Demo</code> but your <code>base</code> tag is set to <code>demo</code> with a lowercase <code>d</code>:</p>
<pre><code class="lang-auto">&lt;base href="https://GajananVS.github.io/demo/"&gt;
</code></pre>
<p>Can you try setting this to t he following instead and seeing if it makes a difference?</p>
<pre><code class="lang-auto">&lt;base href="https://GajananVS.github.io/Demo/"&gt;
</code></pre>

**Comment by RaminMammadzada** [comment]
<p>I also had the same kind of issue.<br>
Instead of    I wrote .<br>
I changed it to  and it worked.</p>

**Comment by dann-chan** [comment]
<p>I had the same issue too. I just started react (newbie on react). Here is how I fixed it.</p>
<p>I went into the index.html file that I uploaded (aside from the fact that github made it so difficult and not user friendly to upload) I simply used the web edit on github and made live changes.<br>
For my case: my page is <a href="http://username.github.io/Home" rel="noopener nofollow ugc">username.github.io/Home</a> (Home is my repository)<br>
I’ve noticed that when I compiled my react page, it didnt know Home is part of the file directory for my css and js files. Therefore I had to find all the places where it uses</p>
<p>link href="/static/css/main.a124****.chunk.css" or script src="/static/js/2.22a******.chunk.js<br>
and simply add /Home infront.</p>
<p>Therefore,<br>
link href="/Home/static/css/main.a124****.chunk.css"  and script src="/Home/static/js/2.22a******.chunk.js</p>
<p>And where ever your code calls js or css or whatever, you will need to add your repo name infront.</p>
<p>Good luck if you have a huge website. Maybe you should add before uploading</p>

**Reply by doehyunbaek** [reply]
Worked like a charm! If anyone wants a commit example, refer to [this commit](https://github.com/doehyunbaek/deadlines/commit/146e01f4c9d098d6096635c0a111c561f1d9b21b)

**Comment by kaushalzod** [comment]
<p>This worked for angular as well just added repo link and added repo name infront of js files</p>

**Comment by kreshby** [comment]
<p>I solved this issue. It was very simple … but I tried more than 3 hours :rofl:<br>
you just do that</p>
<pre><code class="lang-auto">link href="./Home/static/css/main.a124****.chunk.css" 
</code></pre>
<pre><code class="lang-auto">src="./static/js/2.22a******.chunk.js
</code></pre>
<p>just <strong>dot before /</strong></p>

**Reply by moonsol124** [reply]
thank you very mmmmmmmuch!!!!!!! you saved my three hours....

**Reply by exposir** [reply]
Thanks for the answer, It works, but not elegant enough.

**Comment by BoisAuLit** [comment]
<p>Just in case someone is still having the same problem. I replaced https with http in the homepage property in package.json, and then it just works.</p>

**Comment by dattap647** [comment]
Help me to deploy app


![Screenshot (471)](https://user-images.githubusercontent.com/69711651/209641151-2aa4d591-7987-42d9-97ae-2fd78756f2ef.png)

**Reply by otter-computer** [reply]
Hi @dattap647, I believe that this is a react app, correct? It's likely (though not guaranteed!) that you've not set the `homepage` variable in your `package.json` according to the [react instructions for deploying to Pages](https://create-react-app.dev/docs/deployment/#github-pages). In your case I believe it needs to be set to the following:

```
"homepage": "https://dattap647.github.io/eyeCare"
```
Once you do this, rebuild, then commit and push your site to GitHub I believe it will start working!

**Comment by exposir** [comment]
I think it's a bug in github or compatibility is not good enough, I don't understand why gihub hasn't dealt with this problem yet, this problem is very misleading and makes developers think it's their problem so much that they waste a lot of time.

Until gihub has a good solution, the ideal solution is to use vercel or netlify

**Comment by Chauhdri** [comment]
I also faced the same issue, my website works great while locally and shows a blank page after deployment on live server. I got the hint to sort out this issue by testing react app. I imported React from react in my two components of the app, after that it works fine on the live server too.

**Comment by Versiani-R** [comment]
In my case, my website was loading a 404 next error page, but using next.js, my index page inside the pages directory was called "Index.ts" which resulted in a "Index.html" file, and next couldn't locate the index.html page. I changed directly into the public branch ( which is my deploy branch ) to "index.html" and it worked

**Comment by kovalen1618** [comment]
A little late to the party, but if anyone is using Vite with React and are trying to deploy to GitHub Pages then follow [this article](https://dev.to/rashidshamloo/deploying-vite-react-app-to-github-pages-35hf).

After you follow everything and redeploy your code, you may have to fiddle on GitHub a little by saving page deploy to another branch and then going back to gh-pages or you may just have to be patient.

Hopefully this helps!

**Reply by nataliamelissas** [reply]
The comment in that article to change "/" to "./" inside vite.config.ts worked for me.

**Comment by victor-gargur** [comment]
I have the solution for Angular! On your index.html in the your build doc, ajuste the base href="link"

If your link is href="https://user-name.github.io/" add your repo name in the end of this, like:

from:
base
      href="https://victor-gargur.github.io/"
    /

to:
base
      href="https://victor-gargur.github.io/angular-system-bootstrap-localstorage"
    /

In the index.html of your build page.

**Comment by astrit** [comment]
For any one experiencing similar problem, try to create a action that generates a static site to deploy from the branch works for me ✌️

**Comment by TwelfthS** [comment]
If anyone still struggling with this like I just did, my solution was simple - in index.js, change BrowserRouter like this:

    <BrowserRouter basename='/RepositoryName'>
        <App />
    </BrowserRouter>

Important note: I noticed that even after redeploying with changes, the page was still referring to old code, and I only realized this after I tried to add react independent changes. I ended up deleting the repository and creating a new one, but that shouldn't be necessary, just make sure your fixes are actually applied. Might be some caching problem

**Comment by Subash-MASA** [comment]
Help me to this error
![Screenshot (896)](https://github.com/user-attachments/assets/e9062a02-d80a-4a45-a474-4fe080eb9c6c)

**Comment by ZDanial-44** [comment]
Help me to this error pls i use vite and vue 
<img width="1920" height="1200" alt="Снимок экрана (697)" src="https://github.com/user-attachments/assets/582df826-9ae7-46bd-8dc2-81a1f2c2fab5" />  But with link https://zdanial-44.github.io/School_9_SKO/public i see error  404
File not found

The site configured at this address does not contain the requested file.

**Reply by DevHanza** [reply]
This will help you: https://youtu.be/hn1IkJk24ow
